src/p2p/Join/index.ts

```diff
@@ -251,17 +259,35 @@ export function updateRecord(txs: P2P.JoinTypes.Txs, record: P2P.CycleCreatorTyp
 
   if (config.p2p.useJoinProtocolV2) {
     // for join v2, add new standby nodes to the standbyAdd field ...
-    for (const standbyNode of drainNewJoinRequests()) {
-      record.standbyAdd.push(standbyNode)
+    for (const request of txs.standbyAdd) {
+      const publicKey = request.sign.owner
+      const node = NodeList.byPubKey.get(publicKey)
+      if (node) {
+        record.standbyAdd.push(request)
+      } else {
+        /* prettier-ignore */ if(logFlags.important_as_error) warn(`join:updateRecord:standbyAdd: node not found: ${publicKey}`)
+      }
     }
 
     // ... and unjoining nodes to the standbyRemove field ...
-    for (const publicKey of drainNewUnjoinRequests()) {
-      record.standbyRemove.push(publicKey)
+    for (const request of txs.standbyRemove) {
+      const publicKey = request.sign.owner
+      const node = NodeList.byPubKey.get(publicKey)
```

Comment:
This line wont work for standbyRemove. standby nodes are not on the Nodelist.byPubKey array, so this will always return false. the map that you want to use is the standbyNodesInfo one or whatever its exact name is